jupyterhub/spawners.py

```diff
@@ -259,20 +268,21 @@ def get_pod_manifest(self):
         ]
         self.volume_mounts.append(volume_mount)
 
-        # 5. Configure autosaving script execution hook
-        self.lifecycle_hooks = {
-            "preStop": {
-                "exec": {
-                    "command": [
-                        "/bin/sh",
-                        "-c",
-                        "/usr/local/bin/pre-stop.sh",
-                        "||",
-                        "true",
-                    ]
+        # 5. Configure autosaving script execution hook if we are not using a persistent volume
+        if not options.get("pvc_name"):
```

Comment:
Supporting both approaches (PV and autosave) may require a few other changes. Currently, the UI fetches the autosave branches directly from GitLab.

We could add an endpoint to provide the information about an existing volume for a specific commit, or we can move all the autosave/volumes management to renku-notebooks. That requires a new endpoint to provide a list of the commits having an autosave for a target project.

I prefer this second option because it's cleaner, it masks unnecessary deployment details (branches vs PV), and it would make it simple to add another option in the UI to discard the previously saved work when starting a new pod.

Reply:
so the API would return generically "there is some way of recovering unsaved work for these commits" regardless of whether it's an autosave branch or an existing PV?

Reply:
Exactly.

Reply:
I haven't addressed this in this PR because I think it needs some API design - but I opened an issue and I suggest we discuss the design and tackle next sprint: https://github.com/SwissDataScienceCenter/renku-notebooks/issues/584